### DISCUSSION TITLE
AI

### DISCUSSION THREAD

**Discussion by KTechDev4213** [nanoframework/Home#1084] [General]
Is there any plan to add support for any AI library. Like tensorflow lite etc.

**Comment by josesimoes** [comment]
Yes that's on our (long) todo list. 😅

**Reply by KiwiBryn** [reply]
Hi @KTechDev4213 

I do a bit of ML.Net on ARM64 *nix with YoloV5+ONNX for process control so was interested to see if viable a few months ago. 

My approach was to a tool like Edge Impulse to generate code for an executable model. 

Paused project due native interop (me needing to revisit some of my assumptions) + memory management(model used too much) issues

Was also wondering if there was scope for using the ONNX runtime for ARM64 with NF but only so many hours in day

@KiwiBryn

**Comment by KTechDev4213** [comment]


**Reply by josesimoes** [reply]
Would like to discuss this in more detail with you. Are you on our Discord server? If yes, please send me a DM over there. Otherwise send me an email please.

**Comment by KiwiBryn** [comment]
Hi,

Non trivial….

I was training models (accelerometer+ gyro) with Edge Impulse, building then downloading them onto Arduino Nano 33 BLE and STM Boards.

Then as an “extension exercise” I had a go at getting EdgeImpulse to generate different device implementations, then hacking that code so I could build it with the NF tool chain.

Ran out of time but the next step would be using interop to call my DLL from C#.

How much money you got? 😊

@KiwiBryn
